lib/req/steps.ex

```diff
@@ -611,7 +611,8 @@ defmodule Req.Steps do
                 :transport_opts,
                 :proxy_headers,
                 :proxy,
-                :client_settings
+                :client_settings,
+                :hostname
```

Comment:
Thank you for the PR. Note we're explicitly retrieving :hostname option elsewhere:

https://github.com/wojtekmach/req/blob/v0.3.5/lib/req/steps.ex#L618

Given your patch, that code was never correct. Can you remove it? Also, could you write a test for this change?

Reply:
If I'm not mistaken the explicit retrieving of `:hostname` still needs to happen for the option to work. 

The problem I faced was that passing `:hostname` raises on https://github.com/wojtekmach/req/blob/v0.3.5/lib/req/steps.ex#L606 because `:hostname` was not part of the MapSet.